packages/@react-aria/utils/src/number.ts

```diff
@@ -16,3 +16,28 @@
 export function clamp(value: number, min: number = -Infinity, max: number = Infinity): number {
   return Math.min(Math.max(value, min), max);
 }
+
+export function snapValueToStep(value: number, min: number, max: number, step: number): number {
+  let remainder = ((value - min) % step);
+  let snappedValue = Math.abs(remainder) * 2 >= step
+    ? (value - Math.abs(remainder)) + step
+    : value - remainder;
+
+  if (snappedValue < min) {
+    snappedValue = min;
+  } else if (snappedValue > max) {
+    snappedValue = min + Math.floor((max - min) / step) * step;
+  }
```

Comment:
slightly confused here, why are min and max treated differently?
why isn't snappedValue = max for the greater than max case

Reply:
I guess because the max might not be on a step boundary? I copied this from v2

Reply:
but min might also not be a step?

Reply:
Don't the steps start at the min?

Reply:
Verified that the min is always the origin for steps when provided, otherwise it's 0. This is the correct behavior